### PR TITLE
Fix Terraform CKV_AWS_6

### PR DESCRIPTION
**Pull request by Saarett**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Fixed CKV_AWS_6 by actually fetching the instance count as int and not the list and improved the tests

### CODE REVIEW

File: checkov/terraform/checks/resource/aws/ElasticsearchNodeToNodeEncryption.py [hunk 1]
```diff
@@ -23,12 +24,12 @@ def scan_resource_conf(self, conf):
                 if "instance_count" not in cluster_config:
                     return CheckResult.PASSED
                 self.evaluated_keys = ['cluster_config/[0]/instance_count']
-                instance_count = cluster_config["instance_count"]
-                if isinstance(instance_count, int):
+                instance_count = force_int(cluster_config["instance_count"][0])
+                if instance_count is not None:
```
Comment (tronxd) [inline checkov/terraform/checks/resource/aws/ElasticsearchNodeToNodeEncryption.py:28]: Can rewrite as `if instance count:`

File: checkov/terraform/checks/resource/aws/ElasticsearchNodeToNodeEncryption.py [hunk 2]
```diff
@@ -23,12 +24,12 @@ def scan_resource_conf(self, conf):
                 if "instance_count" not in cluster_config:
                     return CheckResult.PASSED
                 self.evaluated_keys = ['cluster_config/[0]/instance_count']
-                instance_count = cluster_config["instance_count"]
-                if isinstance(instance_count, int):
+                instance_count = force_int(cluster_config["instance_count"][0])
```
Comment (tronxd) [inline checkov/terraform/checks/resource/aws/ElasticsearchNodeToNodeEncryption.py:27]: Check the list is indeed not empty with one element in it

File: checkov/terraform/checks/resource/aws/ElasticsearchNodeToNodeEncryption.py [hunk 3]
```diff
@@ -23,12 +24,12 @@ def scan_resource_conf(self, conf):
...
+                if instance_count is not None:
                     if instance_count > 1:
                         self.evaluated_keys.append('node_to_node_encryption/[0]/enabled')
                         if "node_to_node_encryption" in conf.keys() and "enabled" in conf["node_to_node_encryption"][0]:
-                            if conf["node_to_node_encryption"][0]["enabled"]:
+                            if conf["node_to_node_encryption"][0]["enabled"][0]:
```
Comment (tronxd) [inline checkov/terraform/checks/resource/aws/ElasticsearchNodeToNodeEncryption.py:32]: Same here

File: tests/terraform/checks/resource/aws/test_ElasticsearchNodeToNodeEncryption.py [hunk 4]
```diff
@@ -1,166 +1,42 @@
...
-                                 '${var.log_publishing_application_cloudwatch_log_group_arn}']}],
-                         'tags': ['${module.label.tags}'], 'depends_on': [['${aws_iam_service_linked_role.default}']]}
-        scan_result = check.scan_resource_conf(conf=resource_conf)
-        self.assertEqual(CheckResult.PASSED, scan_result)
-
-    def test_success_single_node(self):
-        resource_conf = {'count': ['${var.enabled ? 1 : 0}'], 'domain_name': ['${module.label.id}'],
-                         'elasticsearch_version': ['${var.elasticsearch_version}'],
-                         'advanced_options': ['${var.advanced_options}'], 'ebs_options': [
-                {'ebs_enabled': ['${var.ebs_volume_size > 0 ? True : False}'],
```
Comment (tronxd) [inline tests/terraform/checks/resource/aws/test_ElasticsearchNodeToNodeEncryption.py:137]: Very good refactor, I assume all was transferred to the .tf file in this PR right?

**Review by tronxd — APPROVED** [review]
GJ! had some comments

File: checkov/terraform/checks/resource/aws/ElasticsearchNodeToNodeEncryption.py (hunk 2 above)
Comment (Saarett) [reply checkov/terraform/checks/resource/aws/ElasticsearchNodeToNodeEncryption.py:27]: can it be empty?

File: tests/terraform/checks/resource/aws/test_ElasticsearchNodeToNodeEncryption.py (hunk 4 above)
Comment (Saarett) [reply tests/terraform/checks/resource/aws/test_ElasticsearchNodeToNodeEncryption.py:137]: Thanks! Right.

File: checkov/terraform/checks/resource/aws/ElasticsearchNodeToNodeEncryption.py (hunk 2 above)
Comment (Saarett) [reply checkov/terraform/checks/resource/aws/ElasticsearchNodeToNodeEncryption.py]: @tronxd Please approve changes.

File: checkov/terraform/checks/resource/aws/ElasticsearchNodeToNodeEncryption.py (hunk 3 above)
Comment (Saarett) [reply checkov/terraform/checks/resource/aws/ElasticsearchNodeToNodeEncryption.py:37]: @tronxd Please approve changes.

**Pull request merged**